pandas/core/generic.py

```diff
@@ -1799,6 +1799,13 @@ def __array__(self, dtype=None):
         return com.values_from_object(self)
 
     def __array_wrap__(self, result, context=None):
+        if (context is not None and context[1] is not None and
+                len(context[1]) == 2 and context[1][1] is self and
+                isinstance(context[1][0], np.ndarray)):
+            # TODO: Can we return NotImplemented earlier?  By the time we
+            #  get here we've calculated `result`, which may not be cheap
```

Comment:
@shoyer any idea if there is a more elegant way to do this?

Reply:
> What are you trying to fix here? 

#22537.  In master ndarray[int] + DataFrame[timedelta64[ns]] treats the ndarray as timedelta64 instead of raising TypeError.

Reply:
I'm not familiar with exactly how pandas implements binary ops, but this doesn't look like the right place to fix this.

The problem is likely in the `DataFrame.__radd__` or `DataFrame.__array_ufunc__` method (which NumPy calls instead of `__radd__` if defined, see [here](http://www.numpy.org/neps/nep-0013-ufunc-overrides.html#recommendations-for-implementing-binary-operations) for details and recommendations)

Reply:
Thanks.  Looks like its extra-simple: Series and Index both define `__array_priority__` but DataFrame does not.

Reply:
Wow, I'm kind of surprised we never defined `DataFrame.__array_priority__` :)